### PR TITLE
fix: added support for embedded chats

### PR DESCRIPTION
**Pull request by tlstommy**
This is a continuation of a previous [pull request](https://github.com/SevenTV/Extension/pull/919)

I have added `"all_frames":true`, to the manifest for URLs that match a chat embed or popup to support embedded iframes showing 7tv emotes. 

Per Melonifys suggestion I have made it so that it will only work when it detects chat embeds.

### CODE REVIEW

**Comment by tlstommy** [comment]
this should fix the following issues: https://github.com/SevenTV/Extension/issues/187 https://github.com/SevenTV/Extension/issues/916 https://github.com/SevenTV/Extension/issues/844

**Comment by Feridinha** [comment]
Dead devs xd

**Comment by Quoirky** [comment]
need that

**Comment by tlstommy** [comment]
Hoping it will be approved @Quoirky! I've got some really useful things I've made for twitch chat that would be even better with 7tv lol!

**Comment by Feridinha** [comment]
> Hoping it will be approved @Quoirky! I've got some really useful things I've made for twitch chat that would be even better with 7tv lol!

Same, I think the extension will not be updated anymore, and they will create a new one from scratch 🤷

**Comment by lennartkloock** [comment]
> > > Hoping it will be approved @Quoirky! I've got some really useful things I've made for twitch chat that would be even better with 7tv lol!
> > 
> > 
> > Same, I think the extension will not be updated anymore, and they will create a new one from scratch 🤷
> 
> Hopefully not, I imagine that they are just undergoing restructuring due to new teams/devs coming in.

I've got to disappoint you. We're currently not planning to continue using or maintaining any of the old code. We're rewriting it completely from scratch in this [monorepo](https://github.com/SevenTV/SevenTV) and fixing literally every bug in the process.
(one of the new devs here btw)

**Comment by tlstommy** [comment]
> > > > Hoping it will be approved @Quoirky! I've got some really useful things I've made for twitch chat that would be even better with 7tv lol!
> 
> > > 
> 
> > > 
> 
> > > Same, I think the extension will not be updated anymore, and they will create a new one from scratch 🤷
> 
> > 
> 
> > Hopefully not, I imagine that they are just undergoing restructuring due to new teams/devs coming in.
> 
> 
> 
> I've got to disappoint you. We're currently not planning to continue using or maintaining any of the old code. We're rewriting it completely from scratch in this [monorepo](https://github.com/SevenTV/SevenTV) and fixing literally every bug in the process.
> 
> (one of the new devs here btw)

Well, in that case I hope you all consider adding this as a feature!

File: manifest.config.ts [hunk 1]
```diff
@@ -50,6 +50,7 @@ export async function getManifest(opt: ManifestOptions): Promise<Manifest.WebExt
 			{
 				matches: ["*://*.twitch.tv/*"],
 				js: ["content.js"],
+				all_frames: true,
```
Comment (TroyKomodo) [inline manifest.config.ts:53]: Does this force a reinstall of the extension?

Comment (tlstommy) [reply manifest.config.ts:53]: No, at least as far as I've seen with testing it locally. It just enables 7tv to work in all twitch frames that aren't the top most frame (In this case it's the embedded iframe)

Is that what you're asking?

Comment (TroyKomodo) [reply manifest.config.ts:53]: image

No, I am unsure if this constitutes as a change that will uninstall everyone's extensions so I'm scared to make it.

Comment (tlstommy) [reply manifest.config.ts:53]: Oh yeah no it won't, at least it shouldn't.

Comment (tlstommy) [reply manifest.config.ts:53]: It should just auto update whenever the extension is updated like normal

Comment (TroyKomodo) [reply manifest.config.ts:53]: I think, we wont fix this on this version and will wait for the rewrite due to be completed within the next few weeks.

**Comment by TroyKomodo** [comment]
Features against this repo wont be implemented, we are going to rewrite the extension on this [repo](https://github.com/SevenTV/SevenTV).

File: manifest.config.ts (hunk 1 above)
Comment (tlstommy) [reply manifest.config.ts:53]: ok, should I just resubmit a pr then or will you guys plan on adding this feature in?

**Pull request closed without merging**